app/v2/event-publisher.js

```diff
@@ -0,0 +1,23 @@
+const { validateEvent } = require('./validate-event')
+const { sendMessage, sendMessageBatch } = require('./messaging')
+
+class EventPublisher {
+  constructor (config) {
+    this.config = config
```

Comment:
I noticed there's no reference to App Insights in V2 event-publisher - will V2 events no longer need to use App Insights?

Reply:
It will but we'll push that logic to the event hub.

So with V1 events, each repo needs to pass it's app insights instance to this library and each event is logged using that config.

Which is fine.  But it makes this a bit more complicated to use for consuming repos also adds a bit of risk of inconsistency if all consuming repos are on different versions of the app insights npm package.

So the intention is, let's log every event to app insights in the service that consumes all the events so we can guarantee it's consistent.

Reply:
Yeah, that makes sense. Clears that up :)